lib/hooks/use-products-list.ts

```diff
@@ -0,0 +1,58 @@
+import { SerializeFrom } from '@remix-run/node';
+import { useEffect, useState } from 'react';
+import { getEcomApi, Product } from '../ecom';
+import { useAppliedProductFilters } from './use-applied-product-filters';
+import { useProductSorting } from './use-product-sorting';
+
+export function useProductsList(
```

Comment:
This hook is confusing. It takes the category slug, so I would expect it to return all products from that category. Oh, but it also takes the `initialTotalCount`. What could that possibly mean? I can understand the initial count, I can understand the total count. I cannot understand the initial total count.

Only after looking at the implementation of the hook can I figure out that the hook doesn't actually return all products from the category, it also takes sorting and filtering params into account. But unlike the category slug, it takes them from the URL.

There's also suspicious:
```
    const [products, setProducts] = useState(initialProducts);
    useEffect(() => {
        setProducts(initialProducts);
    }, [initialProducts]);
```

I can't figure out if it's synchronized with the rest of the state in the hook, like when I've loaded a few pages and initialProducts have changed. Why does the hook care if `initialProducts` have changed.

---

It's not "products list", it "product list". It's always singular when you have a list of something.

Reply:
- updated hook to accept all parameters form outside (filters, sort order, category slug)
- renamed to `useProductSlice`, I think this has more sense in our case
- added jsDoc comment where trying to explane better the purpose of this hook

in general, this is done in such a way, cause we have initial data from loader on first page render, but then if user changes filters or sort order, remix loads a new data from client side (without full page reload) and updates the result of `useLoaderData` hook.  this is why we need this strange `useEffect`